scripts/iscsictl.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) 2015 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import argparse
+import os
+import re
+import sys
+
+import sh
+
+
+# open stdout in unbuffered mode
+sys.stdout = os.fdopen(sys.stdout.fileno(), "wb", 0)
+
+
+class Key(object):
+    """Class used to create and reuse temporal SSH keys."""
+
+    def __init__(self, name=None):
+        """Create a new key without passphrase if there is any."""
+        if not name:
+            self.name = '.iscsi_fake_id_dsa'
+        if not os.path.exists(self.name):
+            sh.ssh_keygen('-t', 'dsa', '-f', self.name, '-N', '')
+
+        os.chmod(self.key(), 0600)
+        os.chmod(self.pub_key(), 0600)
+
+    def key(self):
+        """Return the private key filename."""
+        return self.name
+
+    def pub_key(self):
+        """Return the public key filename."""
+        return self.name + '.pub'
+
+    def clean_key(self):
+        """Remove private and public temporal keys."""
+        if os.path.exists(self.key()):
+            os.remove(self.key())
+            os.remove(self.pub_key())
+
+    # def __del__(self):
+    #     """Remove the key."""
+    #     # Because of the way Python works, there is not guarantee that
+    #     # this method is always called.  For this case it doesn't
+    #     # matter, because the old key will be reused.
+    #     self.clean_key()
+
+
+class SSH(object):
```

Comment:
Why don't you use [Paramiko](http://www.paramiko.org/) for the ssh related stuff?